Examples/testCustomEntitlementsComputation/README.md

```diff
@@ -85,12 +85,56 @@ After calling this method, you might need to call your backend to refresh entitl
 Call `purchase(package:)` through either the Async / Await or completion blocks alternatives:
 
 ```swift
-let (transaction, customerInfo, userCancelled) = try await Purchases.shared.purchase(package: package)
+do {
+    let (transaction, customerInfo, _) = try await Purchases.shared.purchase(package: package)
+    print(
+        """
+        Purchase finished:
+        Transaction: \(transaction.debugDescription)
+        CustomerInfo: \(customerInfo.debugDescription)
+        """
+    )
+} catch ErrorCode.receiptAlreadyInUseError {
+    print("The receipt is already in use by another subscriber. " +
+          "Log in with the previous account or contact support to get your purchases transferred to " +
+          "regain access")
+} catch ErrorCode.paymentPendingError {
+    print("The purchase is pending and may be completed at a later time." +
+          "This can happen when awaiting parental approval or going through extra authentication flows " +
+          "for credit cards in some countries.")
+} catch ErrorCode.purchaseCancelledError {
+    print("Purchase was cancelled by the user.")
+} catch {
+    print("FAILED TO PURCHASE: \(error.localizedDescription)")
+}
 ```
 
 ```swift
 Purchases.shared.purchase(package: package) { (transaction, customerInfo, error, userCancelled) in
-    // code to handle here
+    if let error = error {
+        if error.code == ErrorCode.receiptAlreadyInUseError.rawValue {
```

Comment:
⚠️ this is not a great practice, it's assuming the `error.domain`. Instead I recommend following the `Swift migration guide](https://github.com/RevenueCat/purchases-ios/blob/main/Sources/DocCDocumentation/DocCDocumentation.docc/V4_API_Migration_guide.md#error-handling):
```swift
if let error = error as? ErrorCode {
	switch error {
		case .purchaseInvalidError: ...
		default: ...
	}
```

Reply:
I had a vague recollection about this but didn't know where to look, thanks!